### PR TITLE
[v6r15] Bug fixes and better logging in the Matcher

### PR DESCRIPTION
**Pull request by atsareg**
Fixed a bug resulting in ignoring multiProcessor job requirement for sites presenting an empty Tag list.

Better and prettier logging of the matching parameters

### CODE REVIEW

**Comment by Andrew-McNab-UK** [comment]
Can I clarify what the behaviour will be after this change? 

Currently if I have a batch queue on a CE where the batch queue has the tag Multiprocessor, then single processor jobs without that tag still go there, and multiprocessor jobs with that tag also go to non-tagged queues/CEs elsewhere.

The sites tend to have a multiprocessor queue (e.g. "mcore") alongside a generic queue ("long") for single processor jobs. Will this change get the MP and SP jobs only to the right type of queue?

**Comment by atsareg** [comment]
If a job has a tag requirement Multiprocessor it will only be matched by a pilot which is presenting this tag. So, it should be checked that the queue tag Multiprocessor is picked up by the pilot. And vice versa. If pilot only should take jobs with tag Multiprocessor then it should present a RequiredTag parameter with the value Multiprocessor. In this case the double matching is done. So, both job abd site requirement must be defined.
This change is fixing the bug that when pilot was presenting Tag parameter being an empty list, the job requirement was simply ignored.

**Comment by pigay** [comment]
Review ok

File: Core/Utilities/PrettyPrint.py [hunk 1]
```diff
@@ -9,8 +9,13 @@
 
 import StringIO
 
-def int_with_commas(i):
-  s = str(i)
+def int_with_commas( inputValue ):
+  """ Utility to make a string of a large integer with comma separated degrees
+      of thousand
+  :param int inputValue: value to be interpreted
```
Comment (andresailer) [inline Core/Utilities/PrettyPrint.py]: You need an empty line before the parameter list starts.

**Pull request merged**